Get `error-details` from CI App  and enhance findings with more details

MarkusZoppelt opened the pull request:
### Motivation & Context
Authenticated users can get more extensive information when they are authenticated with a CI App.

### Description
This PR gets error-details from the CI App and enhances findings with more details.

**We also handle connection errors during gracefully during run, so that failed API requests don't stop the fuzzing run.**

JackuB commented on internal/api/error_details.go:
```diff
@@ -2,23 +2,77 @@ package api
+	outdated := false
+	if exists {
+		// check if the file is older than 7 days
+		file, err := os.Stat(errorFile)
```
Wouldn't it be better to store a creation date in the JSON as part of the data structure?

MarkusZoppelt replied:
Hm maybe, but it introduces another place to check (and parse) and update details in the backend. I think we might do that in the future, but maybe not in this PR?

I am not even sure if (and how often) we need to update the downloaded file in the first place...

MarkusZoppelt replied:
I removed the date check since the error-details file is negligible in size

jochil commented:
Did you check how it behaves without internet connection or a slow one? IMO we should set a fast timeout, as we do not want users to wait for the API request when all what they want to do is fuzzing. 
Maybe we should also be graceful with error codes != 200... why stopping the fuzzing run when there might be a problem with the API. WDYT?

Also an integration test for checking for the additional fields/infos would be useful, or?

MarkusZoppelt commented:
> Did you check how it behaves without internet connection or a slow one? IMO we should set a fast timeout, as we do not want users to wait for the API request when all what they want to do is fuzzing.
> Maybe we should also be graceful with error codes != 200... why stopping the fuzzing run when there might be a problem with the API.

@jochil I changed it so that status errors only get logged and don't stop the fuzzing run. I also refactored the API `sendRequest` function to specify a timeout.

jochil reviewed (commented):
It might be the wrong place to discuss, but I am not sure if we should start just copying the example projects to `e2e-tests/samples`... wdyt about symlinking it?

MarkusZoppelt commented on e2e-tests/samples/cmake/cifuzz.yaml:
```diff
@@ -0,0 +1,3 @@
+no-notifications: true
+server: app.staging.code-intelligence.com
```
I removed the entire e2e tests from this PR since it's already pretty big. Will add them in another PR

MarkusZoppelt commented on internal/cmd/run/run.go:
```diff
@@ -791,6 +802,21 @@ Your results will not be synced to a remote fuzzing server.`)
 	return willSync, nil
 }
 
+func (c *runCmd) getErrorDetails() (*[]finding.ErrorDetails, error) {
```
Done!

MarkusZoppelt commented on internal/api/error_details.go:
```diff
@@ -2,35 +2,47 @@ package api
+		return nil, err
+	}
+
+	resp, err := client.sendRequestWithTimeout("GET", url, nil, token, 2*time.Second)
```
What is your suggestion? 3 seconds? 4?

JackuB replied:
Hmm, good question. We have two things to consider - how fast (and resilient) our API is. And how comfortable are we with timing out this request - e.g., is it better for a command to take longer or to include all information?

HenrichN replied:
I would propose to give it time, perhaps the same time which was discussed above (10s). 2s might be too strict in cases with slow network connections on the user's side.

HenrichN commented on internal/api/api.go:
```diff
@@ -224,22 +245,30 @@ func (client *APIClient) StartRemoteFuzzingRun(artifact *Artifact, token string)
 
+// sendRequest sends a request to the API server with a default timeout of 10 seconds.
 func (client *APIClient) sendRequest(method string, endpoint string, body io.Reader, token string) (*http.Response, error) {
+	timeout := 10 * time.Second
```
What about leaving the timeout here to 10s and we off course monitor the performance very closely in the backend. If the backend is under so much pressure that we can't get a response within 10 seconds we also need to heavily invest in it. But perhaps sometimes the user has a slow/unstable connection which might drag it along.
